Impro1834 remove time zone

mspelman07 opened the pull request:
Addresses #1834 

Removes code related to local time zones.
Acceptance tast data on branch impro1834_remove_time_zone

Testing:
 - [x] Ran tests and they passed OK

bayliffe reviewed (changes requested):
Thanks Marcus, this looks good. There are still some bits to tidy up in: these files:
- improver/blending/weighted_blend.py
- improver_tests/blending/weighted_blend/test_WeightedBlendAcrossWholeDimension.py
- improver/developer_tools/metadata_interpreter.py
- improver_tests/developer_tools/test_MOMetadataInterpreter.py
- improver_tests/developer_tools/conftest.py (associated with above)
- improver/metadata/constants/time_types.py
- improver_tests/spotdata/test_SpotExtraction.py
- improver_tests/spotdata/test_build_spotdata_cube.py (modify the comment, retain the test)
- improver_tests/acceptance/test_spot_extract.py (some more acceptance test data to delete from spot-extract).

bayliffe commented on improver/developer_tools/metadata_interpreter.py:
```diff
@@ -92,7 +91,7 @@
 UNBLENDED_TIME_COORDS = ["time", "forecast_period", "forecast_reference_time"]
 BLENDED_TIME_COORDS = ["time", "blend_time"]
-LOCAL_TIME_COORDS = ["time", "time_in_local_timezone"]
+LOCAL_TIME_COORDS = ["time"]
```
You can delete `LOCAL_TIME_COORDS` entirely I believe.

mspelman07 replied:
Quite right. I've removed it entirely.

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -105,36 +101,28 @@ def setUp(self):
+        location_points = np.array([location for location in locations])
+        bounds = np.array([[location - 1, location] for location in locations])
+
         # Broadcast the times to a 2-dimensional grid that matches the diagnostic
```
```suggestion
        # Broadcast the locations to a 2-dimensional grid that matches the diagnostic
```

mspelman07 replied:
updated

mspelman07 replied:
updated everywhere.

mspelman07 replied:
updated everywhere

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -105,36 +101,28 @@ def setUp(self):
         diagnostic_cube_2d_time = diagnostic_cube_yx.copy()
-        diagnostic_cube_2d_time.coord("time").rename("time_in_local_timezone")
-        diagnostic_cube_2d_time.add_aux_coord(self.time_aux_coord, data_dims=(0, 1))
+        diagnostic_cube_2d_time.add_aux_coord(self.location_aux_coord, data_dims=(0, 1))
```
```suggestion
        diagnostic_cube_2d_aux.add_aux_coord(self.location_aux_coord, data_dims=(0, 1))
```
Here and surrounds.

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -105,36 +101,28 @@ def setUp(self):
-        self.expected_spot_time_coord = self.time_aux_coord.copy(
+        points = [self.location_aux_coord.points[y, x] for y, x in expected_indices]
+        bounds = [self.location_aux_coord.bounds[y, x] for y, x in expected_indices]
+        self.expected_spot_time_coord = self.location_aux_coord.copy(
```
```suggestion
        self.expected_spot_aux_coord = self.location_aux_coord.copy(
```
And where it is used elsewhere in this file.

mspelman07 replied:
updated everywhere

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -293,15 +281,16 @@ def test_scalar_and_nonscalar_coords(self):
         scalar, nonscalar = plugin.get_aux_coords(
             self.diagnostic_cube_2d_time, x_indices, y_indices
         )
+        print(nonscalar)
```
Delete

mspelman07 replied:
Deleted remaining errant print statements.

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -311,7 +300,7 @@ def test_multiple_nonscalar_coords(self):
-        additional_2d_crd = self.time_aux_coord.copy()
+        additional_2d_crd = self.location_aux_coord.copy()
         additional_2d_crd.rename("kittens")
         self.diagnostic_cube_2d_time.add_aux_coord(additional_2d_crd, data_dims=(0, 1))
```
```suggestion
        self.diagnostic_cube_2d_aux.add_aux_coord(additional_2d_crd, data_dims=(0, 1))
```
Here and surrounds.

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -293,15 +281,16 @@ def test_scalar_and_nonscalar_coords(self):
         x_indices, y_indices = self.coordinate_cube.data
 
         scalar, nonscalar = plugin.get_aux_coords(
             self.diagnostic_cube_2d_time, x_indices, y_indices
```
```suggestion
            self.diagnostic_cube_2d_aux, x_indices, y_indices
```
You get the idea.

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -337,9 +326,8 @@ def test_coordinate_with_bounds_extraction(self):
-
         points, bounds = plugin.get_coordinate_data(
-            self.diagnostic_cube_2d_time, x_indices, y_indices, coordinate="time"
+            self.diagnostic_cube_2d_time, x_indices, y_indices, coordinate="location"
```
```suggestion
            self.diagnostic_cube_2d_aux, x_indices, y_indices, coordinate="location"
```

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -353,9 +341,9 @@ def test_coordinate_without_bounds_extraction(self):
         x_indices, y_indices = self.coordinate_cube.data
 
-        self.diagnostic_cube_2d_time.coord("time").bounds = None
+        self.diagnostic_cube_2d_time.coord("location").bounds = None
```
```suggestion
        self.diagnostic_cube_2d_aux.coord("location").bounds = None
```

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -353,9 +341,9 @@ def test_coordinate_without_bounds_extraction(self):
+        self.diagnostic_cube_2d_time.coord("location").bounds = None
         points, bounds = plugin.get_coordinate_data(
-            self.diagnostic_cube_2d_time, x_indices, y_indices, coordinate="time"
+            self.diagnostic_cube_2d_time, x_indices, y_indices, coordinate="location"
```
```suggestion
            self.diagnostic_cube_2d_aux, x_indices, y_indices, coordinate="location"
```

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -385,7 +373,7 @@ def test_building_cube(self):
         )
         self.assertArrayEqual(
-            result.coord("time").points, self.expected_spot_time_coord.points
+            result.coord("location").points, self.expected_spot_time_coord.points
```
```suggestion
            result.coord("location").points, self.expected_spot_aux_coord.points
```

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -489,12 +477,13 @@ def test_2d_aux_coords(self):
         retained as 1D coordinates associated with the spot-index on the
         spotdata cube."""
         plugin = SpotExtraction()
+        print(self.diagnostic_cube_2d_time)
```
Delete.

bayliffe commented on improver_tests/spotdata/test_SpotExtraction.py:
```diff
@@ -489,12 +477,13 @@ def test_2d_aux_coords(self):
             new_title="IMPROVER Spot Forecast",
         )
-        self.assertEqual(result.coord("time"), self.expected_spot_time_coord)
+        self.assertEqual(result.coord("location"), self.expected_spot_time_coord)
```
```suggestion
        self.assertEqual(result.coord("location"), self.expected_spot_aux_coord)
```

bayliffe reviewed (changes requested):
Thanks again. Tiny stuff left.

bayliffe reviewed (changes requested):
Marcus, you just need to recreate the acceptance test data checksums following the removal of the spot-extract data. Otherwise this is now fine.

mspelman07 commented:
I'd actually just forgotten to commit the acceptance test data changes around the spot-extract data (but the test were all passing for me so I hadn't noticed). It should all work now if you update the acceptance test branch.

MoseleyS reviewed (changes requested):
Looking good. I think I've found one reference in the documentation that needs reviewing.

mspelman07 commented:
I found one piece of documentation that needed removing. Hopefully it's the same one you were thinking of?

MoseleyS reviewed (changes requested):
You found the right bit. I'm not sure what happened to my comment, it seems to have been lost somewhere along the way.

`numba` and `pytz` now don't have a "Required for CLIs" section. This makes `pytz` seem redundant, but I think there is somewhere it is still used. Can you reference it here. I guess `numba` still does some good, although I know we can't use it on our current HPC.

mspelman07 commented:
From a bit of digging I found that pytz was actually only being used by one file that could be changed to use datetime's timezone so I've changed it now and removed pytz from the environments.
 
numba is used by the ensemble copula coupling file if it is available for faster computation of the interpolations. I've tried to track down the CLI's using this and added them to the dependencies file.

MoseleyS reviewed (approved):
RIP Timezone extraction
